cranelift/codegen/src/isa/riscv64/lower.isle

```diff
@@ -97,6 +97,11 @@
         (high Reg (rv_add high_tmp carry)))
     (value_regs low high)))
 
+;; SIMD Vectors
+(rule 8 (lower (has_type (ty_vec128_int ty) (iadd x y)))
+  (if-let $true (has_v))
```

Comment:
Food-for-thought question on this -- this is likely going to be quite common (everything gated by `has_v`), so one option would be to do what the x64 backend does today where if `enable_simd` is turned on then the code generator refuses to run unless `has_v` is enabled (or in x64's case it's SSE3, SSSE3, and SSE4.1). That being said I'm working to lowering x64's requirement to SSE2, the baseline for the architecture, at which point I was actually hoping on removing the `enable_simd` option altogether.

In any case though wanted to throw this out there if you thought that adding `has_v` everywhere was going to be onerous. I suppose though x64 also asserts, for each emitted instruction, that the required instruction sets are all enabled so that would probably need to be added to the risc-v backend to double-check that the `has_v` guards, while not present, still didn't lead to accidentally emitting a vector instruction.

Reply:
Checking `has_v` everywhere is quite verbose. But I'm also in favor of deleting the `enable_simd` setting. (And `enable_float`!)

I was planning on adding a `ty_vec_fits_in_reg` extractor after this PR, to enable some more vector types (`i8x{2,4,..}`), I think restricting these instructions there would make some sense? i.e. if you don't have V, no vector type fits in a register.

Having that extension assert at emit time sounds like a good idea either way.

Reply:
Sounds reasonable to me!